migrate_data_to_hs_access.py

```diff
@@ -0,0 +1,469 @@
+            cur2.execute("""select privilege_id from user_access_to_resource where user_id=%s
+                        and resource_id=%s and assertion_user_id=%s""", (uid, res_id, uid))
+            if cur2.rowcount <= 0:
+                cur2.execute("""insert into user_access_to_resource values (DEFAULT, %s, %s, %s, %s, DEFAULT)""",
+                       (uid, res_id, privilege_id, uid))
+                con2.commit()
+
+    # retrieve all ref_ts resources view_users permissions from Django db and insert them into HSAccess db
+    cur1.execute('select ref_ts_reftimeseries.short_id, auth_user.username from ref_ts_reftimeseries_view_users, ref_ts_reftimeseries, auth_user '
+                 'where auth_user.username != \'root\' and ref_ts_reftimeseries_view_users.reftimeseries_id=ref_ts_reftimeseries.page_ptr_id '
+                 'and ref_ts_reftimeseries_view_users.user_id=auth_user.id')
+
+    if cur1.rowcount > 0:
+        rows = cur1.fetchall()
+        for row in rows:
+            resource_uuid = row['short_id']
+            uname = row['username']
+
+            # get user id from requesting user login name
+            cur2.execute("select user_id from users where user_login=%s", (uname,))
+            if cur2.rowcount > 1 or cur2.rowcount <= 0:
+                print "More than one record or no record for a specific user login %s" % uname
+                sys.exit(1)
+
+            uid = cur2.fetchone()['user_id']
+
+            # set up resource ownership in user_access_to_resource table
+            cur2.execute("select privilege_id from privileges where privilege_code='ro'")
+            if cur2.rowcount <= 0:
+                print "no record for own privilege_code"
+                sys.exit(1)
+            privilege_id = cur2.fetchone()['privilege_id']
+
+            cur2.execute("select resource_id from resources where resource_uuid=%s", (resource_uuid,))
+            if cur2.rowcount <= 0:
+                print "no record in resources for resource_uuid=%s" % resource_uuid
+                continue
+            res_id = cur2.fetchone()['resource_id']
+
+           # only insert into user_access_to_resource when the owner privilege is not already inserted
+            # this check is needed to remove a wrong previous implementation that adds all owners into edit_users and view_users tables
+            cur2.execute("""select privilege_id from user_access_to_resource where user_id=%s
+                        and resource_id=%s and assertion_user_id=%s""", (uid, res_id, uid))
+            if cur2.rowcount <= 0:
+                cur2.execute("""insert into user_access_to_resource values (DEFAULT, %s, %s, %s, %s, DEFAULT)""",
+                       (uid, res_id, privilege_id, uid))
+                con2.commit()
+
+except psycopg2.DatabaseError, e:
+    print 'Error %s' % e
+    sys.exit(1)
+
+finally:
+    if con1:
+        con1.close()
+    if con2:
+        con2.close()
```

Comment:
no need to close these at the end of a process, they'll be closed in garbage collection anyway.